tests/python_functional/src/syslog_ng_config/syslog_ng_config.py

```diff
@@ -71,19 +77,42 @@ def create_global_options(self, **kwargs):
         self.__syslog_ng_config["global_options"].update(kwargs)
 
     # Sources
-    def create_file_source(self, **kwargs):
-        return FileSource(self.__working_dir, **kwargs)
+    def create_file_source(self, **options):
+        driver_name = "file"
+        self.__set_file_name_option(options, "fsinput.log")
+        file_source = SourceDriver(
+            driver_name=driver_name,
+            driver_resources=self.driver_resources[driver_name],
```

Comment:
Thanks, I will fix this.

Reply:
Unfortunately it would be a copy-paste several times in that cases where we have same driver for sources and destinations eg: file-source and file-destination, and so on.
Because of this I would remain class variable: `self.driver_resources`